courses/views/v2/__init__.py

```diff
@@ -254,15 +254,29 @@ class CourseViewSet(viewsets.ReadOnlyModelViewSet):
     def get_queryset(self):
         """Get the queryset for the viewset."""
 
-        return (
+        queryset = (
             Course.objects.select_related("page")
             .prefetch_related("departments")
             .annotate(count_b2b_courseruns=Count("courseruns__b2b_contract__id"))
             .annotate(count_courseruns=Count("courseruns"))
-            .order_by("title")
             .distinct()
         )
 
+        # If IDs are provided via filter, preserve their order
+        id_filter = self.request.query_params.get("id")
+        if id_filter:
+            try:
+                ids = [int(id_str.strip()) for id_str in id_filter.split(",")]
+                queryset = queryset.filter(id__in=ids).order_by(
+                    Case(*[When(id=pk, then=pos) for pos, pk in enumerate(ids)])
```

Comment:
nit: we can use postgres array_position like:
```suggestion
                    Func(ids, F("id"), function="array_position")
```

It will reduce execution time slightly and will also reduce the query size.